File: lksh/2017/games/c.py
import sys

sys.stdin = open('cruel.in')
sys.stdout = open('cruel.out', 'w')
n = int(input())

dp = [0] * (n + 1)
# dp = [0, 1, 1, 2, 0]
for i in range(1, n + 1):
    set_ = set()
    for j in range(i):
        if j <= 1:
            left = 0
        else:
            left = j
        if j >= i - 2:
            right = 0
        else:
            right = i - j - 1
        set_.add(dp[left] ^ dp[right])
        # print(left, right)
    for j in range(n + 1):
        if j not in set_:
            dp[i] = j
            break
# print(*dp)
if dp[n] == 0:
    print('Mueller')
else:
    print("Schtirlitz")
    for j in range(i):
        if j <= 1:
            left = 0
        else:
            left = j
        if j >= n - 2:
            right = 0
        else:
            right = n - j - 1
        if dp[left] ^ dp[right] == 0:
            print(j + 1)

File: lksh/2017/games/c.cpp
#include <bits/stdc++.h>
using namespace std;

#define rep(i, n) for (int i = 0; i < n; i++)
#define ea(i, v) for (auto i : v)

using ll = long long;
int main(){
    freopen("cruel.in", "r", stdin);
    freopen("cruel.out", "w", stdout);
    int n;
    cin >> n;
    vector<int> prev(n + 1);
    vector<int> dp(n + 1);
    for (int i = 1; i <= n; i++){
        unordered_set<int> set_;
        for (int j = 0; j < i; j++){
            int left = (j <= 1 ? 0 : j), right = (j >= i - 2 ? 0 : i - j - 1);
            set_.insert(dp[left] ^ dp[right]);
        }
        for (int j = 0; j <= n; j++){
            if (!set_.count(j)) {
                dp[i] = j;
                break;
            }
        }
    }
    if (!dp[n]){
        cout << "Mueller\n";
    }
    else{
        cout << "Schtirlitz\n";
        for (int j = 0; j < n; j++){
            int left = (j <= 1 ? 0 : j), right = (j >= n - 2 ? 0 : n - j - 1);
            // cout << left << ' ' << right << ' ' << (dp[left] ^ dp[right]) << '\n';
            if ((dp[left] ^ dp[right]) == 0) cout << j + 1 << '\n';
        }
    }
}